main/main.c

```diff
@@ -1779,6 +1779,36 @@ int php_request_startup(void)
 }
 /* }}} */
 
+static void map_ptr_reset_request_strings(void)
+{
+	/* See GH-8646: https://github.com/php/php-src/issues/8646
+	 *
+	 * Interned strings that hold class entries can get a corresponding slot in map_ptr for the CE cache.
+	 * map_ptr works like a bump allocator: there is a counter which increases to allocate the next slot in the map.
+	 *
+	 * For class name strings in non-opcache we have:
+	 *   - on startup: permanent + interned
+	 *   - on request: interned
+	 * For class name strings in opcache we have:
+	 *   - on startup: permanent + interned
+	 *   - on request: either not interned at all, which we can ignore because they won't get a CE cache entry
+	 *                 or they were already permanent + interned
```

Comment:
Or we get new interned+permanent strings (e.g. in `ADD_INTERNED_STRING` in `zend_persist_calc.c`).

Reply:
Yes you're correct, I'll add this line soon-ish. Thanks for checking.